### PR TITLE
Print the first non repeating character in a running stream of characters

### PR DESCRIPTION
**Pull request by gayatri517**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

Code added in cp(competitive programming) folder.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4357

### CODE REVIEW

**Comment by gayatri517** [comment]
@HarshCasper Kindly review it.
Thank you.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Shift your code to some other function and use main for driver code.

**Comment by gayatri517** [comment]
@ankitaggarwal23 I have made the changes. Kindly review it
Thank you.

File: C-Plus-Plus/cp/first_non_repeating_character.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+is encountered you need to keep printing the first non repeating character. */
+
+#include <bits/stdc++.h>
+using namespace std;
+
+/*Program description: 
+First it takes the input from the user, It pushes the character into the queue
+and increments its respective digit. Now, while the queue is not empty, if the 
+frequency of that character is more than 1, it pops it out of the queue
+Else, it is the first occurance and therefore it prints it. */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/first_non_repeating_character.cpp]: It has been repeated twice

File: C-Plus-Plus/README.md [hunk 2]
```diff
@@ -106,7 +106,7 @@
 - [First Negative Integer in Every k sized window](cp/1st_negative_no_in_every_window_of_size_k.cpp)
 - [Rotating an NxN 2D array in 90 degrees anticlockwise direction](cp/rotate2darray.cpp)
 - [Count occurence of Anagram](cp/count_occurences_of_anagrams.cpp)
-
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Remove this line

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -106,7 +106,7 @@
...
+- [First non repeating character in a running stream](cp/first_non_repeating_character.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md:113]: Give a space after this line

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by gayatri517** [comment]
@HarshCasper I have made the changes requested. Kindly review it.
Thank you.

**Comment by gayatri517** [comment]
@HarshCasper please take a look at my PR.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**